Src/FluentAssertions/Specialized/GenericAsyncFunctionAssertions.cs

```diff
@@ -90,5 +90,108 @@ public GenericAsyncFunctionAssertions(Func<Task<TResult>> subject, IExtractExcep
                 return new AndWhichConstraint<GenericAsyncFunctionAssertions<TResult>, TResult>(this, task.Result);
             }
         }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Task{T}"/> does not throw any exception.
+        /// </summary>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public new async Task<AndWhichConstraint<GenericAsyncFunctionAssertions<TResult>, TResult>> NotThrowAsync(string because = "", params object[] becauseArgs)
```

Comment:
I had to read the original issue to understand why all of these new methods were necessary. Would be nice to make that a bit clearer in the PR description.